src/SleepRepository.js

```diff
@@ -0,0 +1,41 @@
+import Sleep from '../src/Sleep'
+
+class SleepRepository{
+  constructor(data) {
+    this.sleep = data.map(data => new Sleep(data))
+  }
+
+  getAvgQuality() {
+    const average = this.sleep.reduce((currentTotal, sleep) => {
+      return sleep.sleepQuality + currentTotal
+    }, 0) / this.sleep.length
+    return Math.round(average * 10)/10
+  }
+
+  filterByUser(userID) {
+    return this.sleep.filter(data => data.userID === userID )
+  }
+
+  getUserAvg(userID, key) {
+    const userSleep = this.filterByUser(userID)
+    const average = userSleep.reduce((currentTotal, sleep) => {
+      return sleep[`${key}`] + currentTotal
```

Comment:
would you mind explaining , in line 22, 29 and 37 ` return` what is the key ? is it the value of the user in case?